src/reducer.js

```diff
@@ -0,0 +1,40 @@
+const initialState = {
+  id: 100,
+  restaurant: {},
+  restaurants: [],
+};
+
+const reducers = {
+  updateRestaurant(state, action) {
+    const { name, value } = action.payload;
+
+    return {
+      ...state,
+      restaurant: {
+        name,
+        value,
+      },
+    };
```

Comment:
```suggestion
    return {
      ...state,
      restaurant: {
        ...state. restaurant,
        [name]: value,
      },
    };
```

리턴 해주는 restaurant 정보가 잘못된것 같아요~ input 태그에서 사용한 name과 restaurant name이 겹치게 사용되고 있어서 주의해서 작업해주세요~